lib/ARMBase.ml

```diff
@@ -86,7 +104,7 @@ let parse_list =
        regs)
 
 let parse_reg s =
-  try Some (List.assoc s parse_list)
+  try Some (List.assoc (String.uppercase s) parse_list)
```

Comment:
String.uppercase has long been deprecates and is no longer available in recent ocaml systems
```suggestion
  try Some (List.assoc (Misc.uppercase s) parse_list)
```